sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala

```diff
@@ -516,18 +516,11 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
    * Checks if the database with the specified name exists.
    */
   override def databaseExists(dbName: String): Boolean = {
-    // To maintain backwards compatibility, we first treat the input is a simple dbName and check
-    // if sessionCatalog contains it. If no, we try to parse it, resolve catalog and namespace,
-    // and check if namespace exists in the catalog.
-    if (!sessionCatalog.databaseExists(dbName)) {
-      val plan = UnresolvedNamespace(parseIdent(dbName))
-      sparkSession.sessionState.executePlan(plan).analyzed match {
-        case ResolvedNamespace(catalog: SupportsNamespaces, ns) =>
-          catalog.namespaceExists(ns.toArray)
-        case _ => true
-      }
-    } else {
-      true
+    val plan = DescribeNamespace(UnresolvedNamespace(resolveNamespace(dbName)), false)
+    sparkSession.sessionState.executePlan(plan).analyzed match {
```

Comment:
Doesn't this throw an `AnalysisException` if the database is missing?

Reply:
Originally, I thought `AnalysisException` won't be threw when the database is missing since [spark/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala](https://github.com/apache/spark/blob/9c21238ce0e6bec676213f4bad4bc1bf4ac16932/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala#L67-L68) only checks whether Catalog exists but I am wrong. There are other cases that may throw an `AnalysisException` and we should handle it